src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,48 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.UserNotAllowed;
 import core.basesyntax.model.User;
+import java.util.HashSet;
+import java.util.Set;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMAL_LOGIN_LENGTH = 6;
+    private static final int MINIMAL_USER_AGE = 18;
+    private static final int MINIMAL_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
+    private Set<String> loginSet = new HashSet<>();
+
+    private void loginSetInit() {
+        for (User user : Storage.people) {
+            loginSet.add(user.getLogin());
+        }
+    }
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws UserNotAllowed {
+        if (user != null) {
+            loginSetInit();
+            if (loginSet != null && loginSet.contains(user.getLogin())) {
+                throw new UserNotAllowed("User with login "
+                        + user.getLogin() + " already exist");
```

Comment:
Checking `loginSet != null` is unnecessary as `loginSet` is initialized on line 16 and never set to null.